pkg/ccl/changefeedccl/encoder_json.go

```diff
@@ -524,12 +577,71 @@ func inSet[S ~string](k S, set map[S]struct{}) bool {
 	return ok
 }
 
+func (e *jsonEncoder) makeValueSchema(updated, prev cdcevent.Row) (json.JSON, error) {
+	ck := tableIDAndVersionPair{
+		{},
+		{tableID: updated.TableID, version: updated.Version, familyID: updated.FamilyID},
+	}
+	if e.beforeField && prev.IsInitialized() {
+		ck[0] = tableIDAndVersion{
+			tableID: prev.TableID, version: prev.Version, familyID: prev.FamilyID,
+		}
+	}
+	if v, ok := e.valueSchemaCache.Get(ck); ok {
+		return v.(json.JSON), nil
+	}
+
+	opts := avro.EnvelopeOpts{
+		BeforeField:        e.beforeField,
+		AfterField:         true,
+		UpdatedField:       e.updatedField,
+		MVCCTimestampField: e.mvccTimestampField,
+		OpField:            true,
+		TsField:            true,
```

Comment:
It seems like these two fields are always set to true currently. Are there eventually use cases where you expect them to be set to false?

Reply:
They are always set to true in the enriched envelope, and always false in other ones.